projects/core/src/auth/client-auth/services/client-error-handling.service.ts

```diff
@@ -26,6 +27,7 @@ export class ClientErrorHandlingService {
     next: HttpHandler
   ): Observable<any> {
     return this.clientTokenService.refreshClientToken().pipe(
+      filter(isNotUndefined),
```

Comment:
Would it be best to make clientTokenService.refreshClientToken not emit undefined?

Maybe it's all the same in the end.  It's just that I noticed that you added `| undefined` to the return value of `refreshClientToken` and in practice, `undefined` is not a value the caller considers.